SofaKernel/modules/SofaCore/src/sofa/core/MultiVecId.h

```diff
@@ -63,12 +61,11 @@ struct StateVecAccessor<DataTypes, V_COORD, V_WRITE>
 {
 public:
     typedef TVecId<V_COORD, V_WRITE> MyVecId;
-    typedef Data<typename DataTypes::VecCoord> MyDataVec;
 
     StateVecAccessor(State<DataTypes>* state, const MyVecId& id) : state(state), id(id) {}
     operator MyVecId() const {  return id;  }
-    const MyDataVec* read()  const {  return state-> read(id);  }
-    MyDataVec* write() const {  return state->write(id);  }
+    auto read()  const {  return state-> read(id);  }
+    auto write() const {  return state->write(id);  }
```

Comment:
I'm not a fan of "hiding" the return type of functions. I think it messes with code readability.
Is it really necessary and/or useful?

Reply:
Without the auto you need to keep the dependency to Data.